how-it-works/2_chainkeytech/04_chain_key_tokens.subpage.md

```diff
@@ -0,0 +1,66 @@
+
+We next give an overview on how chain-key tokens function based on a native chain integration and advanced cryptography.
+
+### Architecture
+
+The architecture of any chain-key token on ICP is using the following building blocks as foundation:
+
+1. Some form of native integration with another blockchain that hosts the underlying token, e.g., the Bitcoin network and its only token, Bitcoin, must be available. This integration must allow canisters on ICP to query balances of addresses of the underlying token on its native chain as well as send transactions to the underlying chain. For underlying chains that use the UTXO accounting model like Bitcoin, the UTXOs of addresses must be queryable as well as those are required to assemble transactions. This integration with the native chain must be done such that no intermediaries are required, i.e., is completely decentralized and realized on chain.
+1. A chain-key implementation of the signature scheme used for signing transactions on the blockchain hosting the underlying asset must be available, e.g., chain-key ECDSA for Bitcoin. This allows for signing transactions for the chain of the underlying token fully on ICP using threshold cryptography without involving an intermediary.
+
+Those two building blocks comprise the native integration with a particular blockchain. ICP has already been integrated with the Bitcoin network using this approach, an integration with Ethereum is planned. Based on this foundation, chain-key token implementations can be built at the smart contract layer. For a specific underlying token, e.g., Bitcoin, this requires a *chain-key token ledger* smart contract, for which the ICRC-1 token standard should be used. The same open source ledger code base can be used and parameterized for any chain-key token to be deployed. In addition to the chain-key token ledger, a chain-key token smart contract is required. This canister essentially replaces the off-chain intermediary of the traditional wrapping architecture with on-chain functionality for the chain-key token architecture. This canister is also called *minter* as it is responsible for creating (and removing) supply of the chain-key token based on in- and outflowing underlying tokens. It keeps any underlying tokens it receives in on-chain custody as long as corresponding chain-key tokens are in circulation, thus ensuring a 1:1 backing of the chain-key token with the underlying asset. The 1:1 backing of the chain-key token by its underlying token can be verified by anyone by checking that the Bitcoin UTXOs the minter claims to hold indeed are held by Bitcoin addresses controlled by the minter, thereby further enhancing trust.
```

Comment:
```suggestion
Those two building blocks comprise the native integration with a particular blockchain. ICP has already been integrated with the Bitcoin network using this approach, and an integration with Ethereum is planned. Based on this foundation, chain-key token implementations can be built at the smart-contract layer. For a specific underlying token, e.g., Bitcoin, this requires a *chain-key token ledger* smart contract, for which the ICRC-1 token standard should be used. The same open source ledger code base can be used and parameterized for any chain-key token to be deployed. In addition to the chain-key token ledger, a chain-key token smart contract is required. This canister essentially replaces the off-chain intermediary of the traditional wrapping architecture with on-chain functionality for the chain-key token architecture. This canister is also called *minter* as it is responsible for creating (and removing) supply of the chain-key token based on in- and out-flowing underlying tokens. It keeps any underlying tokens it receives in on-chain custody as long as corresponding chain-key tokens are in circulation, thus ensuring a 1:1 backing of the chain-key token with the underlying asset. The 1:1 backing of the chain-key token by its underlying token can be verified by anyone by checking that the Bitcoin UTXOs the minter claims to hold indeed are held by Bitcoin addresses controlled by the minter, thereby further enhancing trust.
```

Reply:
It is not quite correct that you need to separate canisters. That's just the ckBTC architecture.
Obviously, this architecture makes sense but it is certainly feasible to have a single canister that handles everything (conversely, having more than 2 canisters may also make sense).

Reply:
Mentioned this below, but rephrased here as it really implied that it must be two canisters.